File: lib/models/grade_item.dart
class GradeItem {
  final String subject;

  /// The display grade string, e.g.:
  ///   "Добър (4)"            — normal редовна grade
  ///   "Добър (4) (П)"        — final grade that came from a поправка/ликвидационна
  ///   "Няма оценка"
  final String grade;
  final String color;

  GradeItem({
    required this.subject,
    required this.grade,
    required this.color,
  });

  bool get isSemester => subject.trim().isEmpty;
  bool get notSemester => !isSemester;
}

File: old_project/GradeItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace e_student
{
    public class GradeItem
    {
        public string Subject { get; set; }
        public string Grade { get; set; }
        public string Color { get; set; }
        public bool IsSemester =>
            string.IsNullOrWhiteSpace(Subject);
        public bool NotSemester =>
            !string.IsNullOrWhiteSpace(Subject);
    }
}
